README.rst

```diff
@@ -20,33 +20,85 @@ configurations remain synchronized with the state of all databases in the
+
 Concepts
 ========
 
+PGShovel has a few core concepts which are important to understanding it:
+
 Cluster
 -------
 
-A cluster contains replication sets.
+A Cluster contains a collection of Replication Sets.
 
 Replication Set
 ---------------
 
-Replication sets contain a collection of tables that are replicated together.
+Replication Sets contain configuration defining a collection of tables that are replicated together.
+
+Mutation Log
+------------
+
+The Mutation Log is where database mutation events are buffered before being forwarded by
+the Relay to their final destination. Internally, PGShovel uses the PgQ tool to queue these mutations inside the database.
+
+Batch Operations
+----------------
+
+The smallest unit of data in PGShovel.  A Batch Operation is a message describing an action describing one of 4 possible operations:
+
+1. BeginOperation
+2. MutationOperation
+3. CommitOperation
+4. RollbackOperation
+
+Ordered sequences of these operations describe with transactional awareness the mutations made against the tables contained in a Replication Set.
 
 Relay
 -----
 
-The relay consumes batches of mutations for the replication set and forwards
-those mutation batches to their destination.
+Relays are responsible for extracting queued raw mutations made against PostgreSQL (stored internally via PgQ), and translating those raw mutation into Batch Operations and logging them to their final destination.  There is exactly one Relay for each Replication Set.
 
-.. todo:: Document replication worker, loader, target.
+Replication
+-----------
 
-Usage
-=====
+The Replication system is a collection of tools which handle replicating a stream of Batch Operations into another Target, i.e. Cassandra or S3.
+
+Within Replication there are 4 main concepts:
+
+Loader
+~~~~~~
+
+The Loader is responsible for providing an interface to dump the initial data from the tables contained in a Replication Set.
+
+Stream
+~~~~~~
+Abstraction over a stream of BatchOperations, handling internal Replication validation and tooling.
+
+Target
+~~~~~~
+
+Object which models and coordinates replication to a final data store, or target (i.e. Cassandra, S3).  The Target reads in all initial replicated records from the loader and applies them to the target data store, coordinating picking up continued replication (and from that point on forever) via the Stream once able to.
+
+Worker
+~~~~~~
+
+The Worker is a command line daemon responsible for configuring and running the Target, using a configured Stream and Loader.
```

Comment:
Hmmm I might disagree with the comma, but I think I'm going to reword this anyways.